src/rpc/request.cpp

```diff
@@ -167,6 +167,25 @@ void JSONRPCRequest::parse(const UniValue& valRequest)
     // Parse id now so errors from here on will have the id
     id = request.find_value("id");
 
+    // Check for JSON-RPC 2.0 (default 1.1)
+    m_json_version = JSONRPCVersion::V1_LEGACY;
+    const UniValue& jsonrpc_version = request.find_value("jsonrpc");
+    if (!jsonrpc_version.isNull()) {
+        if (!jsonrpc_version.isStr()) {
+            throw JSONRPCError(RPC_INVALID_REQUEST, "jsonrpc field must be a string");
+        }
+        // The "jsonrpc" key was added in the 2.0 spec, but some older documentation
+        // incorrectly included {"jsonrpc":"1.0"} in a request object, so we
+        // maintain that for backwards compatibility.
```

Comment:
In commit "rpc: identify JSON-RPC 2.0 requests" (2ca1460ae3a7217eaa8c5972515bf622bedadfce)

I think it would be a little clearer to say "continue to accept that" instead of "maintain that." Otherwise it sounds like we are trying to maintain incorrectly including the field, not just allowing it if it is specified.